src/main/java/core/basesyntax/service/ParserImpl.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dto.Transaction;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ParserImpl implements Parser {
+    private static final String SPLITTER = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_NAME_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private static final String OPERATION_BALANCE = "b";
+    private static final String OPERATION_SUPPLY = "s";
+    private static final String OPERATION_RETURN = "r";
+    private static final String OPERATION_PURCHASE = "p";
+    private static final int NUMBER_OF_ELEMENTS_TO_SKIP = 1;
+    private DataValidator validator = new DataValidatorImpl();
```

Comment:
le'ts pass DataValidator instance through the constructor - we should not hardcode the implementation here